### DISCUSSION TITLE
AuditV2 中获取审批用户的疑问

### DISCUSSION THREAD

**Discussion by nieen** [hhyo/Archery#2827] [Q&A]
在旧版本Audit 中，在获取工单当前节点的审批用户时，会根据权限组（A） 和 工单所在资源组过滤，预期的审批用户结果是当前工单所在资源组中有审批权限（A）的用户列表。
现在AuditV2中，获取的审批用户是所有用户中 有审批权限（A）的用户列表。
这个逻辑调整是预期的吗？还是说AuditV2中遗漏了再基于工单所在资源组的逻辑？有了解这部分情况的老哥辛苦帮忙解惑。

**Comment by LeoQuote** [comment]
你好，我是这次重构的作者，和原版的区别的原因为了实现的简单和方便覆盖，如果有造成一些不方便还请谅解。
这次重构的主要目的是提供各种层面的覆盖可能，也就是开发者可以自己提供自己的audit 实现，开发者可以选择覆盖最简单的 https://github.com/hhyo/Archery/blob/073560e91e95dec68084445d6f6abff0faaedd4c/sql/utils/workflow_audit.py#L269

也可以选择覆盖更加底层的其他方法。

欢迎你提交新的PR来修复这里不符合你预期的地方，但如果可以的话还是保留一个途径可以调整其行为，比如一个可以覆盖的方法，或者可以覆盖的flag等等。

**Reply by nieen** [reply]
明白，节后我来提一个pr处理

**Reply by nieen** [reply]
pr 已提交 https://github.com/hhyo/Archery/pull/2831
